modules/p2/ding_list/ding_list.module

```diff
@@ -190,10 +190,72 @@ function ding_list_menu() {
     'page arguments' => array(2),
   ));
 
+  $items['ding-list/login-create'] = array(
+    'title' => 'Login create',
+    'page callback' => array('ding_list_login_create'),
+    'delivery callback' => 'ajax_deliver',
+    'access callback' => TRUE,
+  );
+
+  $items['ding-list/login-attach/%/%'] = array(
+    'title' => 'Login create',
+    'page callback' => array('ding_list_login_attach'),
+    'page arguments' => array(2, 3),
+    'delivery callback' => 'ajax_deliver',
+    'access callback' => TRUE,
+  );
+
   return $items;
 }
 
 /**
+ * Handle list attachment for anonuymous users.
+ */
+function ding_list_login_attach($type, $value) {
+  $commands = array();
+
+  if (!user_is_logged_in()) {
+    $commands[] = ajax_command_ding_user_authenticate('');
+  }
```

Comment:
you could do
return array('#type' => 'ajax', '#commands' => $commands);
here, and do without the 'else' structure.
It's a matter of taste, though... personally, I like the "early return style of coding". See f.ex. http://blog.timoxley.com/post/47041269194/avoid-else-return-early